### PR TITLE
Unify how getFnCallDescriptor is handled

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: packages/core/src/internal/io.js [hunk 1]
```diff
@@ -71,46 +71,51 @@ export function race(effects) {
...
   if (process.env.NODE_ENV !== 'production') {
-    check(fn, is.notUndef, `${meth}: argument fn is undefined`)
+    check(fnDescriptor, is.notUndef, `${callFnEffects.join('/')}: argument fn is undefined`)
   }
 
   let context = null
-  if (is.array(fn)) {
-    ;[context, fn] = fn
-  } else if (fn.fn) {
-    ;({ context, fn } = fn)
```
Comment (shinima) [inline packages/core/src/internal/io.js]: Are we going to drop the support for `{ context, fn }`?

File: packages/core/src/internal/io.js [hunk 2]
```diff
@@ -71,46 +71,51 @@ export function race(effects) {
   return makeEffect(effectTypes.RACE, effects)
 }
 
-function getFnCallDesc(meth, fn, args) {
+const callFnEffects = ['call', 'apply', 'cps', 'fork', 'spawn']
+
+function getFnCallDescriptor(fnDescriptor, args) {
   if (process.env.NODE_ENV !== 'production') {
-    check(fn, is.notUndef, `${meth}: argument fn is undefined`)
+    check(fnDescriptor, is.notUndef, `${callFnEffects.join('/')}: argument fn is undefined`)
```
Comment (restrry) [inline packages/core/src/internal/io.js]: imho message  `call/apply/cps/fork/spawn: argument fn is undefined` seems confusing as user calls a method, not all of them

Comment (Andarist) [reply packages/core/src/internal/io.js]: right, gonna move validation to the effect creators from get `getFnCallDescriptor` itself

File: packages/core/src/internal/io.js (hunk 1 above)
Comment (Andarist) [reply packages/core/src/internal/io.js]: Oh, this is a bummer 😅 had no idea this is publicly supported, gonna have to restore it & write a test for this

**Comment by Andarist** [comment]
I've addressed both your concerns

**Pull request merged**